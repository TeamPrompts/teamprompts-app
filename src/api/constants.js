export const ACTIVE = 'active';
export const ASC = 'asc';
export const AUTHOR = 'Author';
export const DESC = 'desc';
export const EXAMPLES = 'Examples';
export const FITB = 'FITB';
export const FITBS = 'FITBs';
export const ID = 'ID';
export const NAME = 'Name';
export const PROMPTS = 'Prompts';
export const STATUS = 'Status';
export const TAGS = 'Tags';
