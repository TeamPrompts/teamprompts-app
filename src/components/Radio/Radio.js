import React from 'react';

function Radio({ id, mode, onChange, value }) {
  const inputId = `${id}-${value}`;
  return (
    <>
      <input
        checked={mode === value}
        data-testid={inputId}
        id={inputId}
        name={id}
        onChange={onChange}
        type="radio"
        value={value}
      />
      <label className="ml-2" htmlFor={inputId}>
        {value}
      </label>
    </>
  );
}

export default Radio;
