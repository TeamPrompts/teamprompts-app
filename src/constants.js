// XS (4), S (8), M (16), L (32)

export const BLANK = '________________'; // M
export const BLANK_XS = '____';
export const MASTER_DETAIL_V1 = 'MASTER_DETAIL_V1';
export const NONE = 'NONE';
export const SEPARATOR = '__BLANK__';
export const TEAMPROMPTS_APP_EXPERIMENT = 'TEAMPROMPTS_APP_EXPERIMENT';

export const modes = {
  blanks: 'blanks',
  blanksXS: 'blanksXS',
  examples: 'examples',
  input: 'input',
  prompts: 'prompts'
};

export const pageTypes = {
  CategoryPage: 'CategoryPage',
  HomePage: 'HomePage',
  PromptsPage: 'PromptsPage'
};
