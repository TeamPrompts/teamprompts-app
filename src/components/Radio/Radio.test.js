import React from 'react';
import { create } from 'react-test-renderer';
import { modes } from '../../constants';
import Radio from './Radio';

describe('Radio', () => {
  const mockOnChange = jest.fn();

  const scenarios = [
    {
      description: 'blanks',
      props: {
        id: 0,
        mode: modes.blanks,
        onChange: mockOnChange,
        value: modes.blanks
      }
    }
  ];

  scenarios.forEach(({ description, props }) => {
    it(description, () => {
      const tree = create(<Radio {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
