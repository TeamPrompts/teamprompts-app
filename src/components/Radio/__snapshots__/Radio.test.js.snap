// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio blanks 1`] = `
Array [
  <input
    checked={true}
    data-testid="0-blanks"
    id="0-blanks"
    name={0}
    onChange={[MockFunction]}
    type="radio"
    value="blanks"
  />,
  <label
    className="ml-2"
    htmlFor="0-blanks"
  >
    blanks
  </label>,
]
`;
